### PR TITLE
Add macro support for object enum types

### PR DESCRIPTION
**Pull request by aki-ks**
This PR implements the serialization of sealed traits/classes that have case objects as implementations.

```scala
sealed trait CaseObjectEnum
case object Alpha extends CaseObjectEnum
```
`Alpha` compiles e.g. to `{"_t":"Alpha"}`

### CODE REVIEW

File: bson/src/main/scala/org/mongodb/scala/bson/codecs/macrocodecs/CaseClassCodec.scala [hunk 1]
```diff
@@ -71,18 +71,18 @@ private[codecs] object CaseClassCodec {
 
     // Type checkers
     def isCaseClass(t: Type): Boolean = t.typeSymbol.isClass && t.typeSymbol.asClass.isCaseClass && !t.typeSymbol.isModuleClass
+    def isCaseObject(t: Type): Boolean = t.typeSymbol.isModuleClass && t.typeSymbol.asClass.isCaseClass && t.typeSymbol.isModuleClass
```
Comment (GeorgiChochov) [inline bson/src/main/scala/org/mongodb/scala/bson/codecs/macrocodecs/CaseClassCodec.scala]:
```suggestion
    def isCaseObject(t: Type): Boolean = t.typeSymbol.isModuleClass && t.typeSymbol.asClass.isCaseClass
```
This is called twice

Comment (aki-ks) [reply bson/src/main/scala/org/mongodb/scala/bson/codecs/macrocodecs/CaseClassCodec.scala]: Oops, fixed

**Comment by ernestochero** [comment]
Hello, @aki-ks Is there some news for this PR?

**Comment by ernestochero** [comment]
Hello, @aki-ks @GeorgiChochov  Is there some news for this PR?

**Comment by GeorgiChochov** [comment]
> Hello, @aki-ks @GeorgiChochov Is there some news for this PR?

Hi ernestochero, unfortunately you're asking the wrong person. I'm just as interested in this as you are.

**Comment by rozza** [comment]
Apologies this has fallen through the cracks. See: mongodb/mongo-java-driver#809 which has ported it to the 4.x driver.

**Pull request closed without merging**